core/trino-main/src/main/java/io/trino/sql/planner/planprinter/TextRenderer.java

```diff
@@ -247,13 +247,6 @@ private static Map<String, String> translateOperatorTypes(Set<String> operators)
                     "HashBuilderOperator", "Right (build) ");
         }
 
-        if (operators.contains("LookupJoinOperator") && operators.contains("DynamicFilterSourceOperator")) {
```

Comment:
> When do we have the case where Join has a LookupJoinOperator and DynamicFilterSourceOperator but not a HashBuilderOperator ?

It's question which operator is first in the pipeline going from input to output (see `io.trino.sql.planner.planprinter.PlanNodeStatsSummarizer#getPlanNodeStats`). It might be `DynamicFilterSourceOperator` or `HashBuilderOperator` (depending if DF is on and join type [broadcast vs hash]).

However, since `DynamicFilterSourceOperator` stats are negligible, we should skip them